### PR TITLE
Update tabling.pl

### PR DESCRIPTION
**Pull request by friguzzi**
One possibility for avoiding redundant answers is this.

### CODE REVIEW

File: library/tabling.pl [hunk 1]
```diff
@@ -179,7 +179,8 @@
     update/4.
 
 update(M:Wrapper, A1, A2, A3) :-
-    M:'$table_update'(Wrapper, A1, A2, A3).
+    M:'$table_update'(Wrapper, A1, A2, A3),
+    A1\==A3.
```
Comment (JanWielemaker) [inline library/tabling.pl]: Shouldn't this be `A1 =@= A2`, as tabling is only about variants?

Comment (JanWielemaker) [reply library/tabling.pl]: Sorry, `\+ A1 =@= A2`?

Comment (friguzzi) [reply library/tabling.pl]: Yes, better

**Comment by JanWielemaker** [comment]
Merged together with #271, \=@= change and added your program to the test suite.

**Pull request closed without merging**